File: yht0827/1월/1월 11일/튜플.java
import java.util.Comparator;
import java.util.Arrays;
import java.util.HashSet;

class Solution {
	public int[] solution(String s) {
		int[] answer = {};
		String[] str = s.replaceAll("[{}]", " ").trim().split(" , ");
		Arrays.sort(str, new Comparator<String>() {

			@Override
			public int compare(String o1, String o2) {
				return o1.length() - o2.length();
			}

		});

		answer = new int[str.length];
		int idx = 0;
		HashSet<Integer> hashset = new HashSet<>();
		for (String temp : str) {
			String[] str1 = temp.split("[,]");
			for (String temp1 : str1) {
				int a = Integer.parseInt(temp1);
				if (!hashset.contains(a)) {
					answer[idx++] = a;
					hashset.add(a);
				}
			}
		}

		return answer;
	}
}

File: Eunleelee/2024-01/1월 11일/튜플.py
def solution(s):
    answer = []
    # 문자열 s를 정리해서 리스트로 변환
    s = s[0:-2]
    s = s.replace("{", "")
    s = s.replace(",", " ")
    s = s.split('} ')
    s.sort(key=len)  # 길이 순으로 정렬

    for i in s:
        if not answer: # 길이가 1일때 
            answer.append(int(i))
        else:
            # 현재 있는 숫자와 주워진 숫자를 비교해서 추가된 숫자를 answer에 추가
            now = set(map(int, i.split(' ')))
            pre = set(answer)
            a = list(now - pre)
            answer.append(a[0])
    
    return answer

print(solution("{{1,2,3},{2,1},{1,2,4,3},{2}}"))